invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from "react";
+import { withCancel } from "react-invenio-forms";
+import { http } from "../../api/config";
+import { Grid, Message, Container, Loader } from "semantic-ui-react";
+import FeaturedCommunity from "./FeaturedCommunity";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/invenio_communities/i18next";
+
+export default class FeaturedCommunities extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: undefined,
+      error: undefined,
+      isLoading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetch && this.cancellableFetch.cancel();
+  }
+
+  fetchData = async () => {
+    this.setState({ isLoading: true });
+
+    this.cancellableFetch = withCancel(http.get("/api/communities/featured"));
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits, isLoading: false });
+    } catch (error) {
+      this.setState({ error: error, isLoading: false });
+      console.error(error);
+    }
+  };
+
+  displayError() {
```

Comment:
is there any need to split this into a method instead of directly placing in the render function?

Reply:
Not really, I just thought it would be more readable. Just moved it to the render function.